src/LearnMe.Core/Services/Calendar/GoogleCalendar.cs

```diff
@@ -0,0 +1,137 @@
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<bool> CreateEventAsync(
+            CalendarEventDto eventData, 
+            string calendarId = Constants.CalendarId, 
+            string timezone = Constants.Timezone, 
+            bool isRecurringEvent = false,
+            Recurrence period = Recurrence.DAILY,
+            int recurringEventsCount = 5,
+            DateTime? recurUntilDateTime = null)
+        {
+            CalendarEvent newDbEvent = _mapper.Map<CalendarEvent>(eventData);
+
+            _eventBuilder.BuildBasicEventWithDescription(
+                newDbEvent.Title,
+                newDbEvent?.Start,
+                newDbEvent?.End,
+                newDbEvent.Description);
+
+            if (isRecurringEvent)
+            {
+                _eventBuilder.SetRecurrence(period, recurringEventsCount, recurUntilDateTime);
+            }
+
+            Event newCalendarEvent = _eventBuilder.GetEvent();
+                
+            var createdEvent = await _calendarService.Events.Insert(newCalendarEvent, calendarId).ExecuteAsync();
+            newDbEvent.CalendarId = createdEvent.Id;
+
+            return await _repository.InsertAsync(newDbEvent);
+        }
+
+        public async Task<bool> DeleteEventAsync(int id)
+        {
+            var eventToBeDeleted = await _repository.GetByIdAsync(id);
+            if (eventToBeDeleted != null)
+            {
+                await _calendarService.Events.Delete(Constants.CalendarId, eventToBeDeleted.CalendarId).ExecuteAsync();
+            }
+
+            return await _repository.DeleteAsync(id);
+        }
+
+        public async Task<IEnumerable<CalendarEventDto>> GetAllEventsAsync(
+            int eventsPerPage,
+            int pageNumber,
+            string calendarId = Constants.CalendarId)
+        {
+            // Step 1 - synchronize Google calendar with DB
+            //await _synchronizer.SynchronizeDatabaseWithCalendarAsync(_googleCrudAccess, _calendarService, _repository);
+
+            // Step 2 - get all data from DB
+            var eventsResult = await _repository.GetAllAsync(eventsPerPage, pageNumber);
+
+            IList<CalendarEventDto> results = new List<CalendarEventDto>();
+            foreach (var eventResult in eventsResult)
+            {
+                results.Add(_mapper.Map<CalendarEventDto>(eventResult));
+            }
+
+            return results;
+        }
+
+        public async Task<CalendarEventDto> GetEventByIdAsync(int id)
+        {
+            var foundEvent = await _repository.GetByIdAsync(id);
+
+            return _mapper.Map<CalendarEventDto>(foundEvent);
+        }
+
+        public async Task<bool> UpdateEventAsync(int id, CalendarEventDto eventData)
+        {
+            CalendarEvent toUpdateData = _mapper.Map<CalendarEvent>(eventData);
+            toUpdateData.Id = id;
```

Comment:
Tego nie za bardzo rozumiem ale nie mam czasu, by pobierać twój branch i przeglądać kod w ide. Jeśli `Id` jest wpisywane przez użytkownika to nie jestem przekonany do tego rozwiązania. pole `Id` powinno być sterowane przez Ciebie w kodzie albo przez EFa. On to robi sam automatycznie